test/dynamo/test_graph_break_messages.py

```diff
@@ -0,0 +1,221 @@
+# Owner(s): ["module: dynamo"]
+
+import logging
+import unittest
+
+import torch
+import torch._dynamo
+import torch._dynamo.config
+import torch._dynamo.test_case
+from torch._dynamo.exc import Unsupported
+from torch.testing._internal.common_utils import munge_exc
+from torch.testing._internal.logging_utils import LoggingTestCase, make_logging_test
+
+
+def _helper(fn, args, backend="eager"):
+    return lambda: torch.compile(fn, backend=backend, fullgraph=True)(*args)
+
+
+"""
+NOTE Adding tests to this file:
+
+It is good practice to add a minimal repro for each graph break site (i.e. `unimplemented()` call
+to make sure that there aren't any errors that occur when generating graph break messages.
+
+If a graph break message test fails because the graph break no longer repros,
+it is good practice to find a new minimal repro that causes the graph break.
+If this is too much work, it is likely safe to skip/remove the test, assuming
+it was previously passing and the graph break message is not changed.
+However, if you add a new graph break or modify a graph break message, you should
+make sure that there is a test for it.
+"""
+
+
+class GraphBreakMessagesTest(torch._dynamo.test_case.TestCase):
+    maxDiff = None
+
+    def test_dynamic_shape_operator(self):
+        def fn():
+            return torch.nonzero(torch.rand([10, 10]))
+
+        self.assertExpectedInlineMunged(
+            Unsupported,
+            lambda: torch.compile(fn, backend="eager", fullgraph=True)(),
+            """\
+Dynamic shape operator
+  Explanation: Operator `aten.nonzero.default`'s output shape depends on input Tensor data.
+  Hint: Enable tracing of dynamic shape operators with `torch._dynamo.config.capture_dynamic_output_shape_ops = True`
+
+  Developer debug context: aten.nonzero.default
+
+
+from user code:
+   File "test_graph_break_messages.py", line N, in fn
+    return torch.nonzero(torch.rand([10, 10]))""",
```

Comment:
Animesh: from user code stack trace should be before developer debug context?

Jansel: put the most important information last? Usually people need to scroll up.

Brian: distinguish data-dependent size output operators. Dynamic shapes in general don't require this flag to work. data-dependent output shapes. Do.

Reply:
question for @bobrenjc93 should we turn the flag on by default?

Reply:
Yes but similar to other dynamic shapes flags, flipping it may not be trivial. Here's a canary PR I did a month ago: https://github.com/pytorch/pytorch/pull/142105

It is on our roadmap so we do expect it to be done this year.